### PR TITLE
MINIFICPP-1000 - Add proper logging for configuration loading

### PR DESCRIPTION
**Pull request by arpadboda**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/Properties.cpp [hunk 1]
```diff
@@ -122,26 +115,27 @@ bool Properties::parseConfigureFileLine(char *buf, std::string &prop_key, std::s
...
   }
+
+  adjustedFilename += fileName;
+
   char *path = NULL;
 #ifndef WIN32
   char full_path[PATH_MAX];
   path = realpath(adjustedFilename.c_str(), full_path);
 #else
-  path = const_cast<char*>(adjustedFilename.c_str());
```
Comment (phrocker) [inline libminifi/src/Properties.cpp]: was this here to avoid a warning?

File: libminifi/src/Properties.cpp [hunk 2]
```diff
@@ -122,26 +115,27 @@ bool Properties::parseConfigureFileLine(char *buf, std::string &prop_key, std::s
...
-      }
-      adjustedFilename += fileName;
-    }
+  if (NULL == fileName) {
+    logger_->log_error("Configuration file path for %s is a nullptr!", getName().c_str());
+    return;
+  }
+
+  std::string adjustedFilename = getHome();
+  // perform a naive determination if this is a relative path
```
Comment (phrocker) [inline libminifi/src/Properties.cpp]: nice at quick glance this seems to clean up the logic and code.

File: libminifi/src/Properties.cpp (hunk 1 above)
Comment (arpadboda) [reply libminifi/src/Properties.cpp]: Yep, we assigned this to a simple char ptr, but that's now const char ptr.

**Comment by phrocker** [comment]
@arpadboda Looks good. Will run a quick test and merge

**Pull request closed without merging**